core/src/main/java/com/oath/oak/OakComparator.java

```diff
@@ -9,6 +9,7 @@
 
 import java.nio.ByteBuffer;
 
+// IMPORTANT: it shouldn't be assumed that any ByteBuffer position starts at zero
```

Comment:
Maybe say it in a positive way, e.g., ByteBuffer position might be any nonnegative integer